### PR TITLE
Add test type for the case where `separator` is not `string`

### PR DESCRIPTION
**Pull request by hong4rc**
This test will make coverage to 100%

### CODE REVIEW

File: test.js [hunk 1]
```diff
@@ -7,4 +7,6 @@ test('main', t => {
 	t.deepEqual(splitOnFirst('a---b---c', '---'), ['a', 'b---c']);
 	t.deepEqual(splitOnFirst('a-b-c', '+'), ['a-b-c']);
 	t.deepEqual(splitOnFirst('abc', ''), ['abc']);
+	const error = t.throws(() => splitOnFirst('abc', null), TypeError);
```
Comment (sindresorhus) [inline test.js]: You can test both the error type and message directly in the throws assertion.

Comment (hong4rc) [reply test.js]: I was try use:
```js
t.throws(() => splitOnFirst('abc', null), TypeError, 'any string')
```

but it passed.

So can I just test fail or pass, without a message?

```diff
--- const error = t.throws(() => splitOnFirst('abc', null), TypeError);
+++ t.throws(() => splitOnFirst('abc', null), TypeError)
--- t.is(error.message, 'Expected the arguments to be of type `string`');
```

Comment (sindresorhus) [reply test.js]: That is not correct. Read the `t.throws` docs. It's all there.

Comment (hong4rc) [reply test.js]: Thanks, I fixed it

**Comment by sindresorhus** [comment]
Thanks :)

**Pull request merged**